src/community/smart-data-loader/src/main/java/org/geoserver/smartdataloader/data/SmartDataLoaderDataAccessFactory.java

```diff
@@ -0,0 +1,356 @@
+                new JDBCDataStoreFactoryFinder().getFactoryFromType(jdbcDataStoreInfo.getType());
+        JDBCDataStore jdbcDataStore = null;
+        DataStoreMetadata dsm = null;
+        try {
+            // TODO need to review (since it's forcing to get a JDBC datastore based on parameters.
+            // Not sure what happen with JNDI)
+            jdbcDataStore = factory.createDataStore(jdbcDataStoreInfo.getConnectionParameters());
+            DataStoreMetadataConfig config =
+                    new JdbcDataStoreMetadataConfig(
+                            jdbcDataStore,
+                            jdbcDataStoreInfo.getConnectionParameters().get("passwd").toString());
+            dsm = (new DataStoreMetadataFactory()).getDataStoreMetadata(config);
+        } catch (SQLException e) {
+            LOGGER.log(
+                    Level.WARNING,
+                    "Sql exception while retrieving metadata from the DB " + e.getMessage());
+        } catch (Exception e) {
+            throw new RuntimeException("Error retrieving metadata from DB.");
+        }
+
+        if (dsm == null) {
+            // cannot get datastoremetadata connected from which obtain db metadata
+            throw new RuntimeException("Cannot connect to DB with defined parameters.");
+        }
+        DomainModelConfig dmc = new DomainModelConfig();
+        dmc.setRootEntityName(rootEntity);
+        DomainModelBuilder dmb = new DomainModelBuilder(dsm, dmc);
+        DomainModel dm = dmb.buildDomainModel();
+        // apply exclusions to original model
+        DomainModel newDomainModel = ExclusionsDomainModelVisitor.buildDomainModel(dm, exclusions);
+        // release datastore before returning model
+        jdbcDataStore.dispose();
+        return newDomainModel;
+    }
+
+    /**
+     * Helper method that allows to save an xml document representing a mapping in smart-appschema
+     * folder.
+     */
+    private File saveMappingDocument(String pathname, String filename, Document mapping) {
+        // create smart-appschema folder in workspace folder if it does not exists
+        File directory = new File(pathname);
+        if (!directory.exists()) {
+            directory.mkdirs();
+        }
+        // save document on filename
+        File file = null;
+        try {
+            TransformerFactory transformerFactory = TransformerFactory.newInstance();
+            Transformer transf = transformerFactory.newTransformer();
+            transf.setOutputProperty(OutputKeys.ENCODING, "UTF-8");
+            transf.setOutputProperty(OutputKeys.INDENT, "yes");
+            transf.setOutputProperty("{http://xml.apache.org/xslt}indent-amount", "2");
+            DOMSource source = new DOMSource(mapping);
+            file = new File(pathname + filename);
+            StreamResult stream = new StreamResult(file);
+            transf.transform(source, stream);
+        } catch (Exception e) {
+            e.printStackTrace();
```

Comment:
Remove the printStackTrace method, replace is with Java log output.

Reply:
fixed